src/Unused/CLI/Util.hs

```diff
@@ -1,11 +1,28 @@
 module Unused.CLI.Util
     ( resetScreen
+    , withInterruptHandler
     , module System.Console.ANSI
     ) where
 
 import System.Console.ANSI
+import Control.Exception (throwTo)
+import System.Posix.Signals (Handler(Catch), installHandler, keyboardSignal)
+import Control.Concurrent (ThreadId, myThreadId)
+import System.Exit (ExitCode(ExitFailure))
 
 resetScreen :: IO ()
 resetScreen = do
     clearScreen
     setCursorPosition 0 0
+
+withInterruptHandler :: IO () -> IO ()
+withInterruptHandler body = do
+    tid <- myThreadId
+    _ <- installHandler keyboardSignal (Catch (handleInterrupt tid)) Nothing
+    body
+
+handleInterrupt :: ThreadId -> IO ()
+handleInterrupt tid = do
+    resetScreen
+    showCursor
+    throwTo tid $ ExitFailure 130
```

Comment:
@mike-burns I like that a lot - addressed in fbbacd4045c3bbe8529da25cd023dc490d10d7af